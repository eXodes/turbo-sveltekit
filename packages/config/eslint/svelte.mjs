import { rules } from "./rules.mjs";
import prettier from "eslint-config-prettier";
import turbo from "eslint-config-turbo/flat";
import onlyWarn from "eslint-plugin-only-warn";
import svelte from "eslint-plugin-svelte";
import unicorn from "eslint-plugin-unicorn";
import globals from "globals";
import svelteParser from "svelte-eslint-parser";
import ts from "typescript-eslint";

export default ts.config(
  unicorn.configs["flat/recommended"],
  ...ts.configs.recommended,
  ...ts.configs.stylistic,
  ...svelte.configs["flat/recommended"],
  prettier,
  ...svelte.configs["flat/prettier"],
  ...turbo,
  {
    plugins: {
      "only-warn": onlyWarn,
    },

    languageOptions: {
      globals: { ...globals.browser, ...globals.node },

      ecmaVersion: "latest",
      sourceType: "module",

      parser: ts.parser,
      parserOptions: {
        tsconfigRootDir: import.meta.dirname,
        extraFileExtensions: [".svelte"],
      },
    },

    rules: {
      ...rules,
    },
  },
  {
    files: ["**/*.svelte"],

    languageOptions: {
      parser: svelteParser,
      parserOptions: {
        parser: ts.parser,
      },
    },
  }
);
