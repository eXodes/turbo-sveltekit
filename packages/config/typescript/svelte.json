{
  "display": "Svelte",
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "lib": ["esnext", "dom", "dom.iterable"],
    "module": "esnext",
    "moduleDetection": "force",
    "moduleResolution": "bundler",
    "noEmit": true,
    "noUncheckedIndexedAccess": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "verbatimModuleSyntax": true,
    "useDefineForClassFields": true,
    "sourceMap": true,
    "strict": true,
    "target": "esnext"
  }
}
