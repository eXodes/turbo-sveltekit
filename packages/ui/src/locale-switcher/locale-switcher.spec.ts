import LocaleSwitcher from "./locale-switcher.svelte";
import { render } from "@testing-library/svelte";
import { describe, expect, test, vi } from "vitest";

describe("LocaleSwitcher", () => {
  test("should render component", async () => {
    const { container } = render(LocaleSwitcher, {
      browser: false,
      invalidate: vi.fn(),
    });

    expect(container.children).toMatchSnapshot();
  });
});
