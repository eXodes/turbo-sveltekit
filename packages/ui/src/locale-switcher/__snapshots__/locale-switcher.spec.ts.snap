// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LocaleSwitcher > should render component 1`] = `
HTMLCollection [
  <select
    aria-label=""
    class="rounded-lg text-sm focus-visible:border-orange-600 focus-visible:ring-orange-600"
  >
    <option
      value="en"
    >
      EN
    </option>
    <option
      value="ms"
    >
      MS
    </option>
    
  </select>,
]
`;
